lib/puppet/network/http/compression.rb

```diff
@@ -18,6 +18,7 @@ module Active
     # return an uncompressed body if the response has been
     # compressed
     def uncompress_body(response)
+      Puppet.deprecation_warning(_('Puppet::Network::HTTP::Compression is deprecated.'))
```

Comment:
I wasn't sure which module was being used, so I added the deprecation warning to every `uncompress_body` method, and since I was doing that, I also added it to the `compress` methods.